packages/eas-cli/src/credentials/ios/appstore/authenticate.ts

```diff
@@ -144,6 +152,33 @@ async function loginWithUserCredentialsAsync({
 }
 
 export async function authenticateAsync(options: Options = {}): Promise<AuthCtx> {
+  if (options.mode === AuthenticationMode.API_KEY) {
+    return await authenticateWithApiKeyAsync(options);
+  }
+  return await authenticateAsUserAsync(options);
+}
+
+async function authenticateWithApiKeyAsync(options: Options = {}): Promise<ApiKeyAuthCtx> {
+  // Resolve the user credentials, optimizing for password-less login.
+  const ascApiKey = await resolveAscApiKeyAsync(options.ascApiKey);
+  const team = await resolveAppleTeamAsync(options);
+  return {
+    team,
+    authState: {
+      context: {
+        token: new Token({
+          key: ascApiKey.keyP8,
+          issuerId: ascApiKey.issuerId,
+          keyId: ascApiKey.keyId,
+          duration: 1200, // 20 minutes
```

Comment:
I don't like "magic numbers". I'd assign this to a properly-named variable. @wkozyra95 would probably not agree with me though :D